cmd/node_builder.go

```diff
@@ -194,8 +194,8 @@ type NodeConfig struct {
 	ProtocolEvents    *events.Distributor
 	State             protocol.State
 	Resolver          madns.BasicResolver
-	Middleware        network.Middleware
-	Network           network.Network
+	EngineRegistry    network.EngineRegistry
+	UnderlayNetwork   network.Underlay
```

Comment:
```suggestion
	NetworkUnderlay   network.Underlay
```

I think its fine to call struct fields with the Network* prefix, UnderlayNetwork sounds like it is an actual network rather than a component in a network. Lets use NetworkUnderlay in these instances.

Reply:
https://github.com/onflow/flow-go/pull/4664/commits/eb615ba9e1e6011141922c2df864cd23e71fe789